### PR TITLE
xml2js: consolidated Option-types

### PR DESCRIPTION
**Pull request by claasahl**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## xml2js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/xml2js@latest/))
was missing the following properties:
1. ValidationError
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5cd365c87e68b81eef17a07dbfc64158bc7a2836
</p>

File: types/xml2js/index.d.ts [hunk 1]
```diff
@@ -90,42 +92,8 @@ export interface BuilderOptions {
...
-    rootName?: string;
-    xmldec?: XmlDeclarationAttributes;
-    doctype?: any;
-    renderOpts?: RenderOptions;
-    headless?: boolean;
-    chunkSize?: number;
-    cdata?: boolean;
-}
+export type Options = Omit<ParserOptions, "preserveChildrenOrder" | "chunkSize">;
+export type OptionsV2 = ParserOptions & BuilderOptions;
```
Comment (72636c) [inline types/xml2js/index.d.ts:96]: It seems this default omits `allowSurrogateChars`

**Review by 72636c — APPROVED** [review]
🎉

**Comment by sandersn** [comment]
Is this ready to merge @72636c ? Or do you want `allowSurrogateChars` to be added before merging?

File: types/xml2js/index.d.ts (hunk 1 above)
Comment (72636c) [reply types/xml2js/index.d.ts:96]: @claasahl any thoughts on ^? It's no biggie.

**Review by claasahl — COMMENTED** [review]
Turns out that I forgot click "submit" 🙄

File: types/xml2js/index.d.ts (hunk 1 above)
Comment (claasahl) [reply types/xml2js/index.d.ts:96]: The default value for allowSurrogateChars is supposed to be false (or at least falsy) - https://www.npmjs.com/package/xml2js#options-for-the-builder-class

**Comment by rostik-t** [comment]
> I just published [`@types/xml2js@0.4.7` to npm](https://www.npmjs.com/package/@types/xml2js).

The compatibility with previous version was broken because of new typescript requirement (typescript@3.5).
This resulted in build error:
`[exec] ERROR in node_modules/@types/xml2js/index.d.ts(95,23): error TS2304: Cannot find name 'Omit'.`

Could you please fix this or change major version of @types/xml2js in order to save backward compatibility?

**Comment by claasahl** [comment]
I understand the issue @rostik-t 

Unfortunately, the version number in the header (`// Type definitions for node-xml2js 0.4`) is correctly aligned with the actual `xml2js` version (0.4.23).

Is it an option for you to upgrade your TypeScript version?

**Pull request merged**